src/core/core.layouts.js

```diff
@@ -220,189 +382,94 @@ module.exports = {
 		// 9. Tell any axes that overlay the chart area the positions of the chart area
 
 		// Step 1
-		var chartWidth = width - leftPadding - rightPadding;
-		var chartHeight = height - topPadding - bottomPadding;
-		var chartAreaWidth = chartWidth / 2; // min 50%
-
-		// Step 2
-		var verticalBoxWidth = (width - chartAreaWidth) / verticalBoxes.length;
-
-		// Step 3
-		// TODO re-limit horizontal axis height (this limit has affected only padding calculation since PR 1837)
-		// var horizontalBoxHeight = (height - chartAreaHeight) / horizontalBoxes.length;
+		var chartHeight = height - userPadding.height;
+		var chartArea = {
+			w: width - userPadding.width,
+			h: chartHeight
+		};
+		var chartAreaWidth = chartArea.w / 2; // min 50%
+
+		// Steps 2-3
+		var params = {
```

Comment:
I think that `chartArea` is mutated while the rest are read-only constants. What if we take `chartArea` out of this and pass it separately and then call `Object.freeze` on `params`. I think that would help a lot in keeping track of which variables change between function calls.

I also wonder if we need to keep track of `chartArea`. It seems like there's a lot of complexity in keeping `outerDims` and `chartArea` in sync with each other and the boxes at all times. But isn't `chartArea.w` just `outerWidth` - `outerDims.left` - `outerDims.right`? I think it'd be massively simpler if we just had to worry about keeping `outerDims` updated. I think we do a pretty good job keeping `outerDims.left` updated, but there might be a couple places where we skip updating `outerDims.right` and so we'd have to check that we always keep it updated in order to make that work

Reply:
@kurkle I think this is the only comment I had that hasn't been addressed yet. I resolved the rest.

I'd like if we could at least add a comment saying that `w` and `h` are the only mutable properties and the rest are read-only or possibly we could go further as I mentioned above

Besides that, this PR looks good to me. Thank you so much for all the work on this. I'm very excited about it!

Reply:
I'll take another look at this later, because I feel you are right and this part can be simplified.